src/v/security/audit/schemas/tests/ocsf_schemas_test.cc

```diff
@@ -0,0 +1,505 @@
+        std::chrono::system_clock::now().time_since_epoch())
+        .count()};
+    auto authn = sa::authentication(
+      sa::authentication::activity_id::logon,
+      sa::authentication::auth_protocol_id::kerberos,
+      std::move(dst_endpoint),
+      sa::authentication::used_cleartext::yes,
+      sa::authentication::used_mfa ::no,
+      std::move(src_endpoint),
+      sa::severity_id::informational,
+      now,
+      sa::user{default_user});
+
+    auto ser = sa::rjson_serialize(authn);
+
+    const ss::sstring expected{
+      R"(
+{
+"category_uid": 3,
+"class_uid": 3002,
+"metadata": )"
+      + metadata_ser + R"(,
+"severity_id": 1,
+"time": )"
+      + ss::to_sstring(now) + R"(,
+"type_uid": 300201,
+"activity_id": 1,
+"auth_protocol_id": 2,
+"dst_endpoint": )"
+      + rp_kafka_endpoint_ser + R"(,
+"is_cleartext": true,
+"mfa": false,
+"src_endpoint": )"
+      + client_kafka_endpoint_ser + R"(,
+"user": )"
+      + default_user_ser + R"(
+}
+)"};
+
+    BOOST_REQUIRE_EQUAL(ser, ::json::minify(expected));
+}
+
+SEASTAR_THREAD_TEST_CASE(validate_application_lifecycle) {
+    auto now = sa::timestamp_t{
+      std::chrono::duration_cast<std::chrono::milliseconds>(
+        std::chrono::system_clock::now().time_since_epoch())
+        .count()};
+    auto app_lifecycle = sa::application_lifecycle(
+      sa::application_lifecycle::activity_id::start,
+      sa::product{test_product},
+      sa::severity_id::informational,
+      now);
+
+    auto ser = sa::rjson_serialize(app_lifecycle);
+
+    const ss::sstring expected{
+      R"(
+{
+  "category_uid": 6,
+  "class_uid": 6002,
+  "metadata": )"
+      + metadata_ser + R"(,
+  "severity_id": 1,
+  "time": )"
+      + ss::to_sstring(now) + R"(,
+  "type_uid": 600203,
+  "activity_id": 3,
+  "app": )"
+      + test_product_ser + R"(
+}
+)"};
+
+    BOOST_REQUIRE_EQUAL(ser, ::json::minify(expected));
+}
+
+SEASTAR_THREAD_TEST_CASE(validate_increment) {
+    auto now = sa::timestamp_t{1};
+    auto app_lifecycle = sa::application_lifecycle(
+      sa::application_lifecycle::activity_id::start,
+      sa::product{test_product},
+      sa::severity_id::informational,
+      now);
+
+    auto increment_time = sa::timestamp_t{2};
+
+    app_lifecycle.increment(increment_time);
+
+    auto increment_time2 = sa::timestamp_t{3};
+
+    app_lifecycle.increment(increment_time2);
+
+    auto ser = sa::rjson_serialize(app_lifecycle);
+
+    const ss::sstring expected{
+      R"(
+{
+  "category_uid": 6,
+  "class_uid": 6002,
+  "count": 3,
+  "end_time": )"
+      + ss::to_sstring(increment_time2) + R"(,
+  "metadata": )"
+      + metadata_ser + R"(,
+  "severity_id": 1,
+  "start_time": )"
+      + ss::to_sstring(now) + R"(,
+  "time": )"
+      + ss::to_sstring(now) + R"(,
+  "type_uid": 600203,
+  "activity_id": 3,
+  "app": )"
+      + test_product_ser + R"(
+}
+)"};
+
+    BOOST_REQUIRE_EQUAL(ser, ::json::minify(expected));
+}
+
+SEASTAR_THREAD_TEST_CASE(validate_api_activity_hash) {
```

Comment:
1. it doesn't really seem like this need to be a seastar test right?
2. it'd be easy to convert this over to use good test, too now that it's available in the tree

Reply:
1. sure i was just cribbing it from somewhere else
2. "good test"?

Reply:
sorry, i meant `google test` gtest framework.

Reply:
ahhhh ok... well I switched to the BOOST before clarification.  you good with it there?  I can start using gtest moving forward

Reply:
oh yeh its all good